Extensions/PlatformBehavior/tests/platformerobjectruntimebehavior.spec.js

```diff
@@ -822,6 +846,108 @@ describe('gdjs.PlatformerObjectRuntimeBehavior', function () {
+        expect(object.getBehavior('auto1').isOnFloor()).to.be(true);
+        expect(object.getBehavior('auto1').isFalling()).to.be(false);
+        expect(object.getBehavior('auto1').isMoving()).to.be(false);
+
+        // Check that the object follow the platform, even if the
+        // movement is less than one pixel.
+        for (let i = 0; i < 5; ++i) {
+          platform.setPosition(
+            platform.getX() + deltaX,
+            platform.getY() + deltaY
+          );
+          runtimeScene.renderAndStep(1000 / 60);
+          expect(object.getBehavior('auto1').isOnFloor()).to.be(true);
+          expect(object.getBehavior('auto1').isFalling()).to.be(false);
+          expect(object.getBehavior('auto1').isMoving()).to.be(false);
+        }
+        expect(object.getX()).to.be(0 + 5 * deltaX);
+        expect(object.getY()).to.be(-30 + 5 * deltaY);
+      });
+    });
+
+    it('falls from a platform moving down faster than the maximum falling speed', function () {
+      for (let i = 0; i < 30; ++i) {
+        runtimeScene.renderAndStep(1000 / 60);
+      }
+      // Check the object has not moved.
+      expect(object.getY()).to.be(-30);
+      expect(object.getX()).to.be(0);
+      expect(object.getBehavior('auto1').isOnFloor()).to.be(true);
+      expect(object.getBehavior('auto1').isFalling()).to.be(false);
+      expect(object.getBehavior('auto1').isMoving()).to.be(false);
+
+      // Check that the object falls
+      // +1 because it's the margin to check the floor
+      platform.setY(platform.getY() + maxDeltaY + 1);
+      runtimeScene.renderAndStep(1000 / 60);
+      expect(object.getBehavior('auto1').isOnFloor()).to.be(false);
+      expect(object.getBehavior('auto1').isFalling()).to.be(true);
+      expect(object.getBehavior('auto1').isMoving()).to.be(true);
+      expect(object.getY()).to.be.above(-30);
+    });
+
+    it.skip('follows a platform that is slightly overlapping its top', function () {
+      for (let i = 0; i < 30; ++i) {
+        runtimeScene.renderAndStep(1000 / 60);
+      }
+      // Check the object has not moved.
+      expect(object.getY()).to.be(-30);
+      expect(object.getX()).to.be(0);
+      expect(object.getBehavior('auto1').isOnFloor()).to.be(true);
+      expect(object.getBehavior('auto1').isFalling()).to.be(false);
+      expect(object.getBehavior('auto1').isMoving()).to.be(false);
+
+      // the platform is slightly overlapping the top of the object
+      platform.setY(object.getY() - platform.getHeight() + 1);
+      runtimeScene.renderAndStep(1000 / 60);
+      // Check that the object falls
+      expect(object.getBehavior('auto1').isOnFloor()).to.be(true);
+      expect(object.getBehavior('auto1').isFalling()).to.be(false);
+      expect(object.getBehavior('auto1').isMoving()).to.be(false);
+      expect(object.getY()).to.be(platform.getY() - object.getHeight());
+    });
+
+    it.skip('must not follow a platform that is moved over its top', function () {
```

Comment:
This is skipped, whats happens currently?

Reply:
It's put on top of the platform and stays on it. It does that as long as the platform is retrieved by the spacial search. The margin is maxFallingSpeed * timedelta, I guess.